sklearn/inspection/_permutation_importance.py

```diff
@@ -56,18 +58,24 @@ def _calculate_permutation_scores(
         y = _safe_indexing(y, row_indices, axis=0)
         if sample_weight is not None:
             sample_weight = _safe_indexing(sample_weight, row_indices, axis=0)
-    else:
+    elif X_type == "polars":
+        X_permuted = X.clone()  # polars DataFrame does not have `copy`
+    else:  # pandas DataFrame or numpy ndarray
```

Comment:
I feel like "to copy" is something we should have dispatched to our adapters (currently under `_set_output.py`

Reply:
I see, I will learn about that part and update this PR.